mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -528,16 +770,20 @@ propagateVectorizationInfo(TransformMapAttr map,
       if (!input[upperDim].has_value()) {
         break;
       }
+      auto groups = contiguousMerges[{map, transform}];
       int64_t maxLen = input[upperDim]->maxLength;
       int64_t coeff = input[upperDim]->needsCoefficient;
+      int64_t align = input[upperDim]->alignment;
       int64_t stride = 1;
-      for (auto pair :
-           llvm::zip(llvm::reverse(lowerDims), llvm::reverse(params))) {
-        uint32_t lowerDim = std::get<0>(pair);
-        int64_t lowerLen = std::get<1>(pair);
+      size_t dim_pos = lowerDims.size() - 1;
+      for (auto g : llvm::reverse(groups)) {
```

Comment:
I can now see why there's sorts() in the groups. However, the groups should not be sorted in integer order, they should be sorted by their position in the Merge parameters list.

Your code will do subtly wrong things for `d <- Merge{A, B, C}(x@2, y@1, z@0)` where `y` and `z` are contiguous.